db/db_access.c

```diff
@@ -167,7 +167,7 @@ int check_sql_access(struct sqlthdstate *thd, struct sqlclntstate *clnt)
     else
         rc = check_user_password(clnt);
 
-    if (gbl_uses_externalauth && externalComdb2AuthenticateUserMakeRequest) {
+    if (gbl_uses_externalauth && externalComdb2AuthenticateUserMakeRequest && !clnt->admin) {
```

Comment:
Should we also reset rc to 0 for `admin`?

Reply:
want to make change for IAM only, if (though very less probability) user/password is also there along with IAM, I think it should should be fine to fallback on it, otherwise that function will also almost always give success.